src/GameLogic/Entities/Iceblock.ts

```diff
@@ -0,0 +1,110 @@
+import {Entity} from '../Entity';
+import {EntityType, FloorType} from '../Enums';
+import {Direction8, Direction8Utils} from '../Enums/Direction8';
+import {Level} from '../Level';
+import {Fireball} from './Fireball';
+
+export class Iceblock implements Entity {
+	public readonly type: EntityType;
+
+	public x: number;
+
+	public y: number;
+
+	public direction: Direction8;
+
+	public contains: Entity;
```

Comment:
`contains` is suggestive of a boolean value - maybe `containedEntity`.

Also, what about ice blocks which contain nothing?

Reply:
Currently it isn't possible to make iceblocks that contain nothing. I figured this would be a cool game mechanic, to insist something must be frozen to make an ice block.